packages/relay/src/lib/services/hbarLimitService/index.ts

```diff
@@ -255,21 +293,29 @@ export class HbarLimitService implements IHbarLimitService {
   private async isDailyBudgetExceeded(
     mode: string,
     methodName: string,
+    txConstructorName: string,
     estimatedTxFee: number = 0,
     requestIdPrefix?: string,
   ): Promise<boolean> {
     if (this.shouldResetLimiter()) {
       await this.resetLimiter();
     }
+    // note: estimatedTxFee is only applicable in a few cases (currently, only for file transactions).
+    //      In most situations, estimatedTxFee is set to 0 (i.e., not considered).
+    //      In such cases, it should still be false if remainingBudget === 0.
```

Comment:
Again, these types of comments may not provide immediate assistance to the authors and co-authors who implemented the logic. However, over time, the authors may forget the rationale behind their decisions, and new developers might question why the equation was not combined for greater conciseness instead of being spread out. 

Even we, as the original authors, were initially uncertain about the logic involved, as discussed in [this conversation](https://github.com/hashgraph/hedera-json-rpc-relay/pull/2950#discussion_r1751694319). 

Therefore, while comments like these may not be beneficial at present, they will serve to document the intended purpose.